// react-router-mobx/lib/RouterStore.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.default = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _desc, _value, _class, _descriptor;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _mobx = require('mobx');

var _queryString = require('./queryString');

function _initDefineProp(target, property, descriptor, context) {
	if (!descriptor) return;
	Object.defineProperty(target, property, {
		enumerable: descriptor.enumerable,
		configurable: descriptor.configurable,
		writable: descriptor.writable,
		value: descriptor.initializer ? descriptor.initializer.call(context) : void 0
	});
}

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _applyDecoratedDescriptor(target, property, decorators, descriptor, context) {
	var desc = {};
	Object['ke' + 'ys'](descriptor).forEach(function (key) {
		desc[key] = descriptor[key];
	});
	desc.enumerable = !!desc.enumerable;
	desc.configurable = !!desc.configurable;

	if ('value' in desc || desc.initializer) {
		desc.writable = true;
	}

	desc = decorators.slice().reverse().reduce(function (desc, decorator) {
		return decorator(target, property, desc) || desc;
	}, desc);

	if (context && desc.initializer !== void 0) {
		desc.value = desc.initializer ? desc.initializer.call(context) : void 0;
		desc.initializer = undefined;
	}

	if (desc.initializer === void 0) {
		Object['define' + 'Property'](target, property, desc);
		desc = null;
	}

	return desc;
}

function _initializerWarningHelper(descriptor, context) {
	throw new Error('Decorating class property failed. Please ensure that transform-class-properties is enabled.');
}

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var stripQuery = function stripQuery(loc) {
	if (loc.query) {
		var search = loc.search,
		    query = loc.query,
		    other = _objectWithoutProperties(loc, ['search', 'query']);

		var queryString = (0, _queryString.stringify)(query);
		return _extends({}, other, {
			search: queryString && '?' + queryString
		});
	}
	return loc;
};

var RouterStore = (_class = function () {
	function RouterStore() {
		_classCallCheck(this, RouterStore);

		this._prevLoc = {};

		_initDefineProp(this, 'location', _descriptor, this);
	}

	_createClass(RouterStore, [{
		key: '__initial',
		value: function __initial(_ref) {
			var _desc2,
			    _value2,
			    _class3,
			    _this = this;

			var location = _ref.location,
			    history = _ref.history;
			var Loc = (_class3 = function () {
				_createClass(Loc, [{
					key: 'query',
					get: function get() {
						return (0, _queryString.parse)(this.search.slice(1));
					},
					set: function set(query) {
						var queryString = (0, _queryString.stringify)(query);
						this.search = queryString && '?' + queryString;
						return query;
					}
				}]);

				function Loc(loc) {
					_classCallCheck(this, Loc);

					(0, _mobx.extendObservable)(this, loc);
				}

				return Loc;
			}(), (_applyDecoratedDescriptor(_class3.prototype, 'query', [_mobx.computed], Object.getOwnPropertyDescriptor(_class3.prototype, 'query'), _class3.prototype)), _class3);


			this.location = new Loc(location);
			this._prevLoc = location;

			this.history = history;

			history.listen(function (location) {
				if (typeof _this.location === 'string') {
					_this.location = new Loc({});
				}
				_this.location.pathname = location.pathname;
				_this.location.search = location.search;
				_this.location.hash = location.hash;
				_this._prevLoc = location;
			});

			(0, _mobx.autorun)(function () {
				if (typeof _this.location === 'string' || _this.location.search !== _this._prevLoc.search || _this.location.hash !== _this._prevLoc.hash || _this.location.pathname !== _this._prevLoc.pathname) {
					_this.push(_this.location);
				}
			});
		}
	}, {
		key: 'push',
		value: function push(loc, state) {
			return this.history.push(stripQuery(loc), state);
		}
	}, {
		key: 'replace',
		value: function replace(loc, state) {
			return this.history.replace(stripQuery(loc), state);
		}
	}]);

	return RouterStore;
}(), (_descriptor = _applyDecoratedDescriptor(_class.prototype, 'location', [_mobx.observable], {
	enumerable: true,
	initializer: function initializer() {
		return {};
	}
})), _class);
exports.default = RouterStore;
