// react-flow-player/dist/lib/init-flow-player-setup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _noop = require('./noop');

var initFlowPlayerSetup = function initFlowPlayerSetup(props) {
	var adaptiveRatio = props.adaptiveRatio,
	    aspectRatio = props.aspectRatio,
	    autoplay = props.autoplay,
	    bgcolor = props.bgcolor,
	    title = props.title,
	    sources = props.sources,
	    fullscreen = props.fullscreen,
	    keyboard = props.keyboard,
	    live = props.live,
	    poster = props.poster,
	    rtmp = props.rtmp,
	    speeds = props.speeds,
	    volume = props.volume,
	    onError = props.onError,
	    onResume = props.onResume;

	window.flowplayer('#' + props.playerId, {
		clip: {
			sources: sources,
			title: title
		},
		adaptiveRatio: adaptiveRatio,
		aspectRatio: aspectRatio,
		autoplay: autoplay,
		bgcolor: bgcolor,
		fullscreen: fullscreen,
		keyboard: keyboard,
		live: live,
		poster: poster,
		rtmp: rtmp,
		speeds: speeds,
		volume: volume
	}).on('resume', onResume).on('error', onError);

	var intervalBreak = setInterval(function () {
		var item = document.getElementsByClassName('fp-controls');
		if (item) {
			(props.onLoad || _noop.noop)();
			clearInterval(intervalBreak);
		}
	}, 100);
};

exports.default = initFlowPlayerSetup;
