// react-flow-player/dist/prop-types-config.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypesConfig = {
	playerId: _propTypes2.default.string.isRequired,
	playerInitScript: _propTypes2.default.string.isRequired,
	sources: _propTypes2.default.array.isRequired,
	className: _propTypes2.default.string,
	adaptiveRatio: _propTypes2.default.bool,
	aspectRatio: _propTypes2.default.string,
	autoplay: _propTypes2.default.bool,
	bgcolor: _propTypes2.default.string,
	title: _propTypes2.default.string,
	fullscreen: _propTypes2.default.bool,
	keyboard: _propTypes2.default.bool,
	live: _propTypes2.default.bool,
	poster: _propTypes2.default.string,
	rtmp: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.object]),
	speeds: _propTypes2.default.array,
	volume: _propTypes2.default.number,
	hlsPlugin: _propTypes2.default.bool,
	customButton: _propTypes2.default.array,
	seeking: _propTypes2.default.number,
	seekingText: _propTypes2.default.string,
	onError: _propTypes2.default.func,
	onResume: _propTypes2.default.func,
	speedPlugin: _propTypes2.default.bool,
	hlsUrl: _propTypes2.default.string,
	speedUrl: _propTypes2.default.string,
	styleUrl: _propTypes2.default.string
};

exports.default = propTypesConfig;
