// react-persian/lib/PersianJalaliDate.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

var _get = function get(_x, _x2, _x3) { var _again = true; _function: while (_again) { var object = _x, property = _x2, receiver = _x3; desc = parent = getter = undefined; _again = false; if (object === null) object = Function.prototype; var desc = Object.getOwnPropertyDescriptor(object, property); if (desc === undefined) { var parent = Object.getPrototypeOf(object); if (parent === null) { return undefined; } else { _x = parent; _x2 = property; _x3 = receiver; _again = true; continue _function; } } else if ('value' in desc) { return desc.value; } else { var getter = desc.get; if (getter === undefined) { return undefined; } return getter.call(receiver); } } };

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

function _inherits(subClass, superClass) { if (typeof superClass !== 'function' && superClass !== null) { throw new TypeError('Super expression must either be null or a function, not ' + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _momentJalaali = require('moment-jalaali');

var _momentJalaali2 = _interopRequireDefault(_momentJalaali);

var PersianJalaliDate = (function (_Component) {
  _inherits(PersianJalaliDate, _Component);

  function PersianJalaliDate() {
    _classCallCheck(this, PersianJalaliDate);

    _get(Object.getPrototypeOf(PersianJalaliDate.prototype), 'constructor', this).apply(this, arguments);
  }

  _createClass(PersianJalaliDate, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      var en = this.props.en;

      // load persian mode by default if no en mode
      if (!en) {
        _momentJalaali2['default'].loadPersian();
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props;
      var timestamp = _props.timestamp;
      var inputFormat = _props.inputFormat;
      var format = _props.format;
      var children = _props.children;
      var fromNow = _props.fromNow;

      var moment = timestamp ? _momentJalaali2['default'].unix(children) : (0, _momentJalaali2['default'])(children, inputFormat || format);
      var formatted = fromNow ? moment.fromNow() : moment.format(format);
      return _react2['default'].createElement(
        'span',
        null,
        formatted
      );
    }
  }], [{
    key: 'propTypes',
    value: {
      children: _react.PropTypes.string,
      timestamp: _react.PropTypes.boolean,
      inputFormat: _react.PropTypes.string,
      format: _react.PropTypes.string,
      en: _react.PropTypes.boolean,
      fromNow: _react.PropTypes.boolean
    },
    enumerable: true
  }, {
    key: 'defaultProps',
    value: {
      format: 'jYYYY/jM/jD'
    },
    enumerable: true
  }]);

  return PersianJalaliDate;
})(_react.Component);

exports['default'] = PersianJalaliDate;
module.exports = exports['default'];
